### PR TITLE
Phone Auth RFC

### PR DESCRIPTION
**Pull request by TorstenDittmann**
https://github.com/appwrite/rfc/blob/rfc-phone-auth/020-phone-auth.md

### CODE REVIEW

File: 020-phone-auth.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+│      End       │
+│                │
+└────────────────┘
+```
+
+### Providers Endpoints
+
+This addition also needs to have more endpoints, which are needed to see which providers are linked to which account and also delete a provider from a user.
+
+#### `GET: /v1/account/providers`
```
Comment (lohanidamodar) [inline 020-phone-auth.md:115]: Why not simply return providers with user object during account.get?
1. That way we will only need a delete endpoint for the provider in Account and Users service,
2. Account service can add provider during session create if a session already exists
3. we only need add provider endpoint for Users service

File: 020-phone-auth.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+### Completion
+
+This step will compare the token's `$id` and `secret` created in the initialization step. 
+#### `POST: /v1/account/sessions/phone/complete`
+
+The payload must contain the `$id` of the Token and the 6 digit number send to the phone representing the Token's `secret`. If both values are valid, this endpoint will either login an existing user or create a new one if the phone number isn't present on any of the user provider attribute.
+
+### Flow
+```
```
Comment (lohanidamodar) [inline 020-phone-auth.md:74]: I like the Ascii flowchart,
https://asciiflow.com/

File: 020-phone-auth.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+This could potentially be solved with the already existing authentication methods, but when a user is already logged in - will append the used provider to the current user.
+
+### Future possibilities
+
+[future-possibilities]: #future-possibilities
+
+<!-- This is also a good place to "dump ideas", if they are out of scope for the RFC you are writing but otherwise related. -->
+
+This implementation will allow more implementation in the future and also decouple the user account from an E-Mail address.
```
Comment (lohanidamodar) [inline 020-phone-auth.md]:
```suggestion
This implementation will allow more providers implementation in the future and also decouple the user account from an E-Mail address.
```

File: 020-phone-auth.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+[unresolved-questions]: #unresolved-questions
+
+<!-- What parts of the design do you expect to resolve through the RFC process before this gets merged? -->
+
+#### Are the required credentials consitent across multiple Third Party Providers?
+
+#### How de we add new providers to a user?
+
+This could potentially be solved with the already existing authentication methods, but when a user is already logged in - will append the used provider to the current user.
```
Comment (lohanidamodar) [inline 020-phone-auth.md:207]: 1. If request is made from account endpoint and there is an active session, add the provider to the current user
2. For users service, there should be separate endpoint to add the provider

Is there something I don't understand here?

File: 020-phone-auth.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+Write your answer below.
+-->
+
+### Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+<!-- What parts of the design do you expect to resolve through the RFC process before this gets merged? -->
+
+#### Are the required credentials consitent across multiple Third Party Providers?
```
Comment (lohanidamodar) [inline 020-phone-auth.md:203]: We should research among most popular providers at the moment, should make the decision simpler

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Small changes, and few questions. 
Rest looks good to me.

File: 020-phone-auth.md (hunk 1 above)
Comment (TorstenDittmann) [reply 020-phone-auth.md:115]: True, should be part of the user object anyway and we can skip this endpoint.

**Pull request closed without merging**